### PR TITLE
Fix fsa labels

### PR DESCRIPTION
**Pull request by yaozengwei**
This PR fixes the issue when we change the fsa attribute `labels`:
- https://github.com/k2-fsa/k2/issues/1104
- https://github.com/k2-fsa/k2/issues/874
- https://github.com/k2-fsa/k2/issues/978

### CODE REVIEW

File: k2/python/k2/fsa.py [hunk 1]
```diff
@@ -222,6 +222,8 @@ def __init__(
             self.__dict__[name] = dict()
 
         self._tensor_attr['scores'] = _k2.as_float(self.arcs.values()[:, -1])
+        self._tensor_attr["labels"] = self.arcs.values()[:, 2]
+        self.labels_version = self._tensor_attr["labels"]._version
```
Comment (csukuangfj) [inline k2/python/k2/fsa.py:226]: Could you add a comment here explaining why
we need `labels_version`?

You can place the link to this PR https://github.com/k2-fsa/k2/pull/1140
to the comment.

**Comment by yaozengwei** [comment]
By adding an attribute `labels_version` for class `Fsa`, we can check whether the labels has been inappropriately modified. For example:
22061673527903_ pic

File: k2/python/k2/fsa.py (hunk 1 above)
Comment (yaozengwei) [reply k2/python/k2/fsa.py:230]: ok

**Review by csukuangfj — APPROVED** [review]

**Comment by csukuangfj** [comment]
Thanks! Looks good to me.

**Comment by csukuangfj** [comment]
Could you tell us which version of k2 you are using?

(Please include the date info in the version number.)

**Comment by csukuangfj** [comment]
Could you replace
```
with torch.inference_mode():
```
with
```
with torch.no_grad():
```
and retry?

**Comment by csukuangfj** [comment]
> self.labels_version = self._tensor_attr["labels"]._version

As a workaround, you can use
```
self.labels_version = 0
```
if you are sure you won't modify `fsa.labels` once `fsa` is constructed.

---

You can change the file  `fsa.py` in-place.
